tests/functional_tests/util_resources.py

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2021 The Monero Project
+# 
+# All rights reserved.
+# 
+# Redistribution and use in source and binary forms, with or without modification, are
+# permitted provided that the following conditions are met:
+# 
+# 1. Redistributions of source code must retain the above copyright notice, this list of
+#    conditions and the following disclaimer.
+# 
+# 2. Redistributions in binary form must reproduce the above copyright notice, this list
+#    of conditions and the following disclaimer in the documentation and/or other
+#    materials provided with the distribution.
+# 
+# 3. Neither the name of the copyright holder nor the names of its contributors may be
+#    used to endorse or promote products derived from this software without specific
+#    prior written permission.
+# 
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+# MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+# THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+# STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+# THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+"""
+    Help determine how much CPU power is available at the given time
+    by running numerical calculations
+"""
+
+from __future__ import print_function
+import datetime
+import multiprocessing
+import psutil
+from math import pi
+
+def available_ram_gb():
+    ram_bytes = psutil.virtual_memory().available
+    kilo = 1024
+    ram_gb = ram_bytes / kilo / kilo / kilo
+    return ram_gb
+
+def pi_digits_lib(digits):
+    """
+    Returns the number of digits of Pi
+    """
+    return float('{:.{}f}'.format(pi, digits))
+
+def pi_digits_numeric_yield(digits):
+    """Generate x digits of Pi.
+    https://stackoverflow.com/a/54967370
```

Comment:
Using code from stackoverflow is a serious licensing issue. Also, as a practical matter, you can perhaps do something much simpler here, such as a calculation with the decimal or math libraries.

Reply:
The proper copyrights actually belong to these gents:

This is implementing a spigot algorithm by A. Sale, D. Saada, S. Rabinowitz, mentioned on http://mail.python.org/pipermail/edu-sig/2012-December/010721.html

https://gist.github.com/deeplook/4947835

Reply:
Of course algorithms to compute pi are plentiful in the public domain. The issue here is you've now attributed work to stackoverflow and taking that attribution back would be doubly bad. Plus, using an optimized C-based mathematical function, as found in decimal or math, seems maybe a better strategy.

Reply:
I think that as long as the PR isn't merged, I don't give nor take anything. That's what a PR is for. Otherwise you create a destructive mental pressure on contributors, making them refrain from doing any contributions.
Besides, a link is not a copyright clause.

Reply:
I'm not saying that you've broken the project by introducing a PR with an SO link, I'm saying it could be problematic to merge said code, given its history.

Reply:
Then I'm very happy, that I misunderstood you.

The Decimal solution work equally fine.

Reply:
The decimal solution isn't right. The same code runs for 2s on my machine and 2000s on the CI machine. I'll try to avoid Decimal.

Reply:
Trying the Leibnitz' formula now.